ToolBox/AccernDataSplitter/Program.cs

```diff
@@ -0,0 +1,131 @@
+                    var symbol = lineSplit[4];
+
+                    if (month == DateTime.MinValue)
+                    {
+                        month = date;
+                    }
+
+                    //First time the symbol is detected for the month
+                    if (!symbolDictionary.ContainsKey(symbol))
+                    {
+                        symbolDictionary.Add(symbol, new List<string>());
+                    }
+                    
+                    if (month.Month != date.Month && symbolDictionary.ContainsKey(symbol))
+                    {
+                        WriteLinesToDisk(month, symbolDictionary);
+
+                        Log.Trace("Done with month {0} and year {1}", month.Month, month.Year);
+
+                        month = date;
+                    }
+
+                    symbolDictionary[symbol].Add(line);
+                }
+
+                Log.Trace("Done With the file {0}" , new FileInfo(fileName).Name);
+                
+                reader.Close();
+                File.Delete(file);            
+            }
+
+            //At the end of the file write the data that is left
+            WriteLinesToDisk(month, symbolDictionary);
+
+            Log.Trace("Done writing a total of {0} files", _count);
+            Console.WriteLine("Press any key to exit...");
+            Console.ReadKey();
+
+        }
+
+        /// <summary>
+        /// Dumps all the processed lines into appropriate files and zips them
+        /// </summary>
+        /// <param name="month">The month for the processed data</param>
+        /// <param name="symbolDictionary">Dictionary of symbol, List of lines</param>
+        public static void WriteLinesToDisk(DateTime month, Dictionary<string, List<string>> symbolDictionary)
+        {
+            foreach (var symbol in symbolDictionary.Keys)
+            {
+                if (!symbolDictionary[symbol].IsNullOrEmpty())
+                {
+                    if (!Directory.Exists(GenerateDirectoryPath(symbol)))
+                        Directory.CreateDirectory(GenerateDirectoryPath(symbol));
+
+                    var filePath = GenerateFilePath(symbol, month);
+
+                    File.AppendAllLines(filePath, symbolDictionary[symbol]);
+                    Compression.Zip(filePath);
+
+                    if (_count % 10000 == 0 && _count > 0) Log.Trace("Done writing {0} files", _count);
+                    _count++;
+
+                    symbolDictionary[symbol].Clear();
+                }
+            }
+        }
+
+        public static string GenerateDirectoryPath(string symbol)
+        {
+            return Path.Combine(Globals.DataFolder, "news", "accern", symbol.ToLower());
+        }
+
+        public static string GenerateFilePath(string symbol, DateTime date)
+        {
+            return Path.Combine(GenerateDirectoryPath(symbol), date.ToString("yyyyMM") + ".csv");
```

Comment:
Should the file path generation be added to `LeanData` with all the other file generation helper methods? Perhaps the argument could be made that this new data is 'outside' the scope of `LeanData` -- but if it's in LEAN, it's nice to have all the file pathing logic in one place coupled with the unit tests in `LeanDataTests`